lib/ace/mode/latex_parser_test.js

```diff
@@ -0,0 +1,181 @@
+/*
+ LatexParser unit tests
+
+ run with node:
+ ```
+    node latex_parser_test.js
+ ```
+
+ @author: alexandr.gudulin@gmail.com
+*/
+
+if (typeof process !== "undefined") {
+    require("amd-loader");
+}
+
+define(function(require, exports, module) {
+"use strict";
+
+var assert = require("../test/assertions");
+
+var Document = require("../document").Document;
+var LatexParser = require("./latex_parser").LatexParser;
+var Tokenizer = require("../tokenizer").Tokenizer;
+var BackgroundTokenizer = require("../background_tokenizer").BackgroundTokenizer;
+var LatexHighlightRules = require("./latex_highlight_rules").LatexHighlightRules;
+var highlighter = new LatexHighlightRules();
+var myBgTokenizer = new BackgroundTokenizer(new Tokenizer(highlighter.getRules()));
+
+function getDiscussions(doc) {
+    myBgTokenizer.setDocument(doc);
+    myBgTokenizer.start(0);
+    var parser = new LatexParser(myBgTokenizer);
+
+    parser.go(doc, false, true);
+    return parser.getDiscussions();
+}
+
+module.exports = {
+
+    name: "Papeeria::LatexParser",
+
+    "test: empty document": function() {
+        var doc = new Document([]);
+        var discussions = getDiscussions(doc);
+
+        assert.equal(discussions.length, 0);
+    },
+
+    "test: one discussion": function() {
+        var doc = new Document([
+            "context line",
+            "% * <foo@gmail.com> dt1:",
+            "% foo comment"
+        ]);
+        var discussions = getDiscussions(doc);
+
+        assert.equal(discussions.length, 1);
+
+        var discussion = discussions[0];
+        assert.equal(discussion.type, '*');
+        assert.equal(discussion.author, 'foo@gmail.com');
+        assert.equal(discussion.datetime, 'dt1');
+        assert.equal(JSON.stringify(discussion.text), JSON.stringify(['foo comment']));
+        assert.equal(discussion.replies.length, 0);
+    },
+
+    "test: one discussion with reply": function() {
+        var doc = new Document([
+            "context line",
+            "% * <foo@gmail.com> dt1:",
+            "% foo comment",
+            "% ^ <bar@gmail.com> dt2:",

```

Comment:
If we add an empty line(s) between this header and the body of the previous comment then this comment will be ignored. Maybe it has sense to parse such situations? @bardsoftware/papeeria-team what do you think?